--- src/main.rs	2022-10-21 18:42:53.607218019 +0200
+++ src/main.rs	2022-10-21 18:42:49.463183260 +0200
@@ -59,10 +59,13 @@
     textdomain(config::PKGNAME).unwrap();
 
     // Load gresources
-    let res = gio::Resource::load(
-        config::PKGDATADIR.to_owned() + &format!("/{}.gresource", config::APP_ID),
-    )
-    .expect("Could not load resources");
+    let path = &format!(
+        "{}/{}/{}.gresource",
+        std::env::var("SNAP").unwrap(),
+        config::PKGDATADIR.to_owned(),
+        config::APP_ID
+    );
+    let res = gio::Resource::load(path).expect("Could not load resources");
     gio::resources_register(&res);
 
     glib::set_application_name(config::NAME);
