diff -Naur a/data/de.haeckerfelix.Fragments.desktop.in.in b/data/de.haeckerfelix.Fragments.desktop.in.in
--- a/data/de.haeckerfelix.Fragments.desktop.in.in	2022-10-27 09:00:58.913555342 +0200
+++ b/data/de.haeckerfelix.Fragments.desktop.in.in	2022-10-27 09:03:43.039630069 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=@NAME@
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
-Icon=@APP_ID@
+Icon=snap.fragments.@APP_ID@
 Exec=@PKGNAME@
 Terminal=false
 Type=Application
diff -Naur a/data/de.haeckerfelix.Fragments.metainfo.xml.in.in b/data/de.haeckerfelix.Fragments.metainfo.xml.in.in
--- a/data/de.haeckerfelix.Fragments.metainfo.xml.in.in	2022-10-27 09:01:07.145662866 +0200
+++ b/data/de.haeckerfelix.Fragments.metainfo.xml.in.in	2022-10-27 09:04:06.979664799 +0200
@@ -2,6 +2,7 @@
 <!-- Copyright 2022 Felix Häcker -->
 <component type="desktop-application">
   <id>@APP_ID@</id>
+  <icon type="stock">snap.fragments.de.haeckerfelix.Fragments</icon>
   <name>@NAME@</name>
   <summary>A BitTorrent Client</summary>
   <translation type="gettext">@PKGNAME@</translation>
diff -Naur a/src/main.rs b/src/main.rs
--- a/src/main.rs	2022-10-27 09:01:17.557800000 +0200
+++ b/src/main.rs	2022-10-27 09:03:12.599388741 +0200
@@ -58,7 +58,8 @@
 
     // Load app resources
     let path = &format!(
-        "{}/{}/{}.gresource",
+        "{}/{}/{}/{}.gresource",
+        std::env::var("SNAP").unwrap(),
         config::DATADIR,
         config::PKGNAME,
         config::APP_ID
