diff -Naur a/data/de.haeckerfelix.Fragments.desktop.in.in b/data/de.haeckerfelix.Fragments.desktop.in.in
--- a/data/de.haeckerfelix.Fragments.desktop.in.in	2023-10-31 13:54:24.314847831 +0100
+++ b/data/de.haeckerfelix.Fragments.desktop.in.in	2023-10-31 13:55:12.723631286 +0100
@@ -1,7 +1,7 @@
 [Desktop Entry]
 # Translators: Do NOT translate or transliterate this text (this is the application name)!
 Name=@NAME@
-Icon=@APP_ID@
+Icon=snap.fragments.@APP_ID@
 Exec=@PKGNAME@
 Terminal=false
 Type=Application
diff -Naur a/data/de.haeckerfelix.Fragments.metainfo.xml.in.in b/data/de.haeckerfelix.Fragments.metainfo.xml.in.in
--- a/data/de.haeckerfelix.Fragments.metainfo.xml.in.in	2023-10-31 13:54:24.314847831 +0100
+++ b/data/de.haeckerfelix.Fragments.metainfo.xml.in.in	2023-10-31 13:56:17.140676303 +0100
@@ -2,6 +2,7 @@
 <!-- Copyright 2022-2023 Felix Häcker -->
 <component type="desktop-application">
   <id>@APP_ID@</id>
+  <icon type="stock">snap.fragments.@APP_ID@</icon>
   <name>@NAME@</name>
   <summary>A BitTorrent Client</summary>
   <translation type="gettext">@PKGNAME@</translation>
diff -Naur a/src/main.rs b/src/main.rs
--- a/src/main.rs	2023-10-31 13:54:24.322847962 +0100
+++ b/src/main.rs	2023-10-31 13:57:11.061553010 +0100
@@ -59,7 +59,8 @@
 
     // Load app resources
     let path = &format!(
-        "{}/{}/{}.gresource",
+        "{}/{}/{}/{}.gresource",
+        std::env::var("SNAP").unwrap(),
         config::DATADIR,
         config::PKGNAME,
         config::APP_ID
